### PR TITLE
settings: Bump minimum supported version to 20.1

### PR DESCRIPTION
**Pull request by miretskiy**
Bump minimum supported version to 20.1 since work has
started on 20.2

Release notes: None

### CODE REVIEW

**Review by irfansharif — CHANGES_REQUESTED** [review]
@tbg or @andreimatei are better persons to review this, staring at this for a while I'm pretty confused by the comment block right above (which is now obviously stale, but not obvious to me how). So I'm not sure if binaryMinSupportedVersion is supposed to be Version20_1 or VersionStart20_1 because of it.

Do we need to support pre-release 20.1 versions? If so, it needs to be VersionStart20_1, not Version20_1. But I don't understand why we need to support pre-release 20.1 versions, and why the version-upgrade test can't simply go through "the final 20.1 binary" before going to HEAD.

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49590#-:-M8LTsmUCXfW-iP_l10v:b3mv6f4)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by andreimatei — COMMENTED** [review]
I'm not sure either, to be honest. I guess I'd look at what the version upgrade test does, and go with `Version20_1` is we can. The risk is that we've given someone a pre-release binary and they'll try to switch from that to 20.2, but it seems unlikely.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49590#-:-M8LXBzG5CmUimqSv-Ws:b-o684qg)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

**Comment by miretskiy** [comment]
I found comment to be confusing as well.  Looking that the historical changes,  we seem to use version start version sometimes, and sometimes release version...

I have tried using VersionStart20_1; unfortunately the TestClusterVersionMixedVersionTooOld test fails for me when I tried to do that -- see https://github.com/cockroachdb/cockroach/pull/49444

--- FAIL: TestClusterVersionMixedVersionTooOld (0.17s)
    test_server_shim.go:225: ensuring SQL migrations: failed to run migration "add created_by columns to system.jobs": add-jobs-idx: schema change cannot be initiated in this version until the version upgrade is finalized
        (1) attached stack trace

Basically, schema change job that I'm trying to run as an sql migration requires cluster to be fully upgraded from 19.2...  I have tried later versions (such as VersionSchemaChange job) that were between releases.  Same thing: this test failes.  Basically, we need 20.1 or greater to be able to run sql migration.

Perhaps @tbg can shed more light on this.

**Comment by dt** [comment]
I think it is supposed to be `Version20_1` -- the rule is you can upgrade from 20.1.0 or any patch thereafter to 20.2, and that is what setting it to `Version20_1` gets us. Setting it to `VersionStart20_1` seems wrong -- that is basically a 19.2 cluster, since it predated all the versions stamped in the course of developing 20.1 to identify breaking changes and migrations, so we should not be allowing upgrading from there -- it'd be just as unsafe as allowing upgrading from `Version19_2`.

**Review by dt — APPROVED** [review]

**Comment by miretskiy** [comment]
bors r=dt
